ext/mysqli/mysqli.c

```diff
@@ -303,6 +303,10 @@ zval *mysqli_read_property(zend_object *object, zend_string *name, int type, voi
 
 	obj = php_mysqli_fetch_object(object);
 
+	if (type == BP_VAR_IS && !obj->ptr) {
+		return &EG(uninitialized_zval);
+	}
```

Comment:
This was simpler than I expected ;) I don't think it's quite right though, in two directions: First, there are a number of property handlers that don't seem to require a valid connection, for example `link_client_version_read()`. There are also some properties that check for STATUS_INITIALIZED and others for STATUS_VALID. Whether or not a property is available depends on the property, we can't make a blanket decision here.

I think we need to change the signature of the property handlers to return SUCCESS/FAILURE (the current return value is unnecessary) to determine whether the property should be included or not.